cmd/tree-walk.go

```diff
@@ -183,11 +184,13 @@ func doTreeWalk(ctx context.Context, bucket, prefixDir, entryPrefixMatch, marker
 		}
 
 		if strings.HasSuffix(entry, slashSeparator) {
-			leafDir = isLeafDir(bucket, pathJoin(prefixDir, entry))
+			leafDir = true
```

Comment:
'leafDir' should be false by default, so 'isDir' would be able to be set to true if 'entry' has a trailing slash.